### ISSUE TITLE
Errors in test scripts go unnoticed and tests still pass

### ISSUE THREAD

**Issue by pspacek** [latchset/pkcs11-provider#252]
**Describe the bug**
`make check` runs tests and often reports `PASS` even though the individual tests themselves report syntax errors.

**To Reproduce**
1. Compile 7daf31711a3d0ddcd05d14e89ff418eb733fea19.
2. `make check`
3. Check `tests/pubkey-softhsm.log`
4. Check that test system reports `PASS`
5. Check log file for errors (e.g. `tests/pubkey-softhsm.log`)

```log
Executing ./tpubkey

## Export RSA Public key to a file
openssl pkey -in $BASEURI -pubin -pubout -out ${TMPPDIR}/baseout.pub
/home/pspacek/.cache/pikaur/build/pkcs11-provider/src/pkcs11-provider/tests/helpers.sh: line 50: [: -eq: unary operator expected
```

**Expected behavior**
- Well, no errors
- Most importantly, errors should be detected!

**Operating environment (please complete the following information):**
 - OS: Arch
 - Version: Fully updated as of 2023-06-21

**Additional context**
Seems like dd6c532aee8878a3c49225664c23ecd8ae084509 caused the error itself, but ignoring errors in test is probably another problem.

**Comment by simo5** [comment]
I think an actual error in a test still results in an ERROR condition, but I will investigate this helper woes ...

**Comment by pspacek** [comment]
Not on my system :shrug: 

```
PASS: pubkey-softhsm
```

```console
$ grep unary tests/pubkey-softhsm.log 
/home/pspacek/.cache/pikaur/build/pkcs11-provider/src/pkcs11-provider/tests/helpers.sh: line 50: [: -eq: unary operator expected
...
```

**Comment by pspacek** [comment]
Oh, nevermind, only now I realized what you mean by "actual error". Well, maybe, but this does not inspire confidence :trollface:

**Comment by Jakuje** [comment]
I saw these errors before, but I was not completely sure what the helper function is trying to achieve in there and given that it was amending the test results I did not touch that with the last changes I was making. Turns out that the `$2` is not defined in most of the calls to the `ossl` helper so just setting default value for that variable should solve the issue. Proposed fix in #254
